packages/nimble-components/src/wafer-map/modules/experimental/hover-handler.ts

```diff
@@ -0,0 +1,79 @@
+import type { WaferMap } from '../..';
+import { PointCoordinates, WaferMapOriginLocation } from '../../types';
+
+/**
+ * HoverHandler deals with user interactions and events like hovering
+ */
+export class HoverHandler {
+    public constructor(private readonly wafermap: WaferMap) {}
+
+    public mousemove(event: MouseEvent): void {
```

Comment:
Our current patterns elsewhere would suggest instead of providing event handlers on a model class for a consumer object to register with, we instead expose `connect` and `disconnect` implementations (called those names for convention's sake), and then in those implementations you would handle registering/unregistering the events against the already available reference to `waferMap`, and calling the `connect` and `disconnect` methods from the `WaferMap`'s `connectedCallback` and `disconnectedCallback` implementations. Similar to [this](https://github.com/ni/nimble/blob/aeaa29b6db90372e2c2adf3f88275b758f1e0771/packages/nimble-components/src/table/models/virtualizer.ts#L56).

Reply:
Additionally, the ultimate name for the event handler function should likely be of the form of `onMouseMove`.

Reply:
I understand, but in our use case we attach and detach events based on the update type in the wafermap. should we call those connect and disconnect instead?

Reply:
Yeah, I'm not a fan of the current implementation being influenced by the need to switch between the two systems. Honestly, I'd rather we do away with the whole `EventCoordinator` class, as I'm not really sure it's offering a clear utility even from a separation of concerns perspective. Even the `EventCoordinatorCallbacks` interface that is declared isn't even used in that file, and should just move to the `types.ts` file.

I would prefer the `WaferMap` just handle registering/unregistering the events that it has handlers for (in its `connectedCallback` and `disconnectedCallback` implementations), and also move to the `connect`/`disconnect` paradigm I mention, and just call those methods temporarily when the update type is switched.